src/test/java/org/opensearch/neuralsearch/query/HybridQueryIT.java

```diff
@@ -206,6 +206,36 @@ public void testTotalHits_whenResultSizeIsLessThenDefaultSize_thenSuccessful() {
         assertEquals(RELATION_EQUAL_TO, total.get("relation"));
     }
 
+    @SneakyThrows
+    public void testMaxScoreCalculation_whenMaxScoreIsTrackedAtCollectorLevel_thenSuccessful() {
+        initializeIndexIfNotExist(TEST_BASIC_VECTOR_DOC_FIELD_INDEX_NAME);
```

Comment:
The primary concern in the unit test was the behavior of collector during multi segment scenario. Does this initialization make sure of 2 segments at least

Reply:
We mostly write multiple segments scenarios when a results has been impacted by concurrent search is enabled and disabled. In this case the results does not get impacted by that.